internal/operator-metrics/metrics.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operatormetrics
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var namespaceFile = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+type OperatorMetrics struct {
+	kubeClient client.Client
+}
+
+func NewOperatorMetrics(config *rest.Config, scheme *runtime.Scheme) (OperatorMetrics, error) {
+	kubeClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return OperatorMetrics{}, err
+	}
+
+	return OperatorMetrics{
+		kubeClient: kubeClient,
+	}, nil
+}
+
+func (d OperatorMetrics) Start(ctx context.Context) error {
+	rawNamespace, err := os.ReadFile(namespaceFile)
+	if err != nil {
+		return fmt.Errorf("error reading namespace file: %w", err)
+	}
+
+	namespace := string(rawNamespace)
+
+	serviceName := fmt.Sprintf("opentelemetry-operator-controller-manager-metrics-service.%s.svc", namespace)
+
+	sm := monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "opentelemetry-operator-metrics-monitor",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name":    "opentelemetry-operator",
+				"app.kubernetes.io/part-of": "opentelemetry-operator",
+				"control-plane":             "controller-manager",
+			},
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/name": "opentelemetry-operator",
+				},
+			},
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+					Interval:        "30s",
+					Path:            "/metrics",
+					Scheme:          "https",
+					ScrapeTimeout:   "10s",
+					TargetPort:      &intstr.IntOrString{IntVal: 8443},
+					TLSConfig: &monitoringv1.TLSConfig{
+						CAFile: "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt",
```

Comment:
is this the configuration of OCP in-cluster monitoring stack? 

If so shall this feature be enabled only on OCP?